packages/material-ui/src/Dialog/Dialog.test.js

```diff
@@ -1,269 +1,243 @@
+
+  it('should render with a TransitionComponent', () => {
+    const Transition = React.forwardRef(() => <div data-testid="Transition" tabIndex={-1} />);
+    const { getAllByTestId } = render(
+      <Dialog open TransitionComponent={Transition}>
+        foo
+      </Dialog>,
+    );
+
+    expect(getAllByTestId('Transition')).to.have.lengthOf(1);
+  });
 
-  it('should render a Modal with TransitionComponent', () => {
-    const Transition = React.forwardRef(() => <div tabIndex={-1} />);
-    const wrapper = mount(
-      <Dialog {...defaultProps} open TransitionComponent={Transition}>
+  it('calls onEscapeKeydown when pressing Esc followed by onClose after the specified transitionDuration', () => {
+    const onEscapeKeyDown = spy();
+    const onClose = spy();
+    const { getByRole } = render(
+      <Dialog open transitionDuration={100} onEscapeKeyDown={onEscapeKeyDown} onClose={onClose}>
         foo
       </Dialog>,
     );
-    assert.strictEqual(wrapper.find(Transition).length, 1);
+    expect(getByRole('dialog')).to.be.ok;
+
+    getByRole('dialog').click();
+    fireEvent.keyDown(document.activeElement, { key: 'Esc' });
+    expect(onEscapeKeyDown.calledOnce).to.equal(true);
+
+    clock.tick(100);
+    expect(onClose.calledOnce).to.equal(true);
   });
 
-  it('should put Modal specific props on the root Modal node', () => {
-    const onEscapeKeyDown = () => {};
-    const onClose = () => {};
-    const wrapper = mount(
+  it('can ignore backdrop click and Esc keydown', () => {
+    const onClose = spy();
+    const { getByRole } = render(
       <Dialog
         open
-        transitionDuration={100}
-        onEscapeKeyDown={onEscapeKeyDown}
-        onClose={onClose}
         disableBackdropClick
         disableEscapeKeyDown
+        onClose={onClose}
+        transitionDuration={0}
       >
         foo
       </Dialog>,
     );
+    expect(getByRole('dialog')).to.be.ok;
 
-    const modal = wrapper.find(Modal);
-    assert.strictEqual(modal.props().open, true);
-    assert.strictEqual(modal.props().BackdropProps.transitionDuration, 100);
-    assert.strictEqual(modal.props().onEscapeKeyDown, onEscapeKeyDown);
-    assert.strictEqual(modal.props().onClose, onClose);
-    assert.strictEqual(modal.props().disableBackdropClick, true);
-    assert.strictEqual(modal.props().disableEscapeKeyDown, true);
-  });
+    getByRole('dialog').click();
+    fireEvent.keyDown(document.activeElement, { key: 'Esc' });
+    expect(onClose.callCount).to.equal(0);
 
-  it('should spread custom props on the paper (dialog "root") node', () => {
-    const wrapper = mount(
-      <Dialog {...defaultProps} data-my-prop="woofDialog">
-        foo
-      </Dialog>,
-    );
-    const modal = wrapper.find(Modal);
-    assert.strictEqual(modal.props()['data-my-prop'], 'woofDialog');
+    clickBackdrop(document.body);
+    expect(onClose.callCount).to.equal(0);
   });
 
-  it('should fade down and make the transition appear on first mount', () => {
```

Comment:
Removed. The fade portion wasn't actually tested. Just that a component with that name was used. `appear` as a prop is an implementation detail. Will investigate what happens if we remove it and provide a better test case.